index.js

```diff
@@ -1,46 +1,69 @@
+require("./messages");
+var steam = require("steam");
+
+const DOTA_APP_ID = 570;
+
 var EventEmitter = require('events').EventEmitter,
     fs = require("fs"),
+    path = require("path"),
     util = require("util"),
     bignumber = require("bignumber.js"),
-    Schema = require('protobuf').Schema,
-    base_gcmessages = new Schema(fs.readFileSync(__dirname + "/generated/base_gcmessages.desc")),
-    gcsdk_gcmessages = new Schema(fs.readFileSync(__dirname + "/generated/gcsdk_gcmessages.desc")),
-    dota_gcmessages = new Schema(fs.readFileSync(__dirname + "/generated/dota_gcmessages.desc")),
-    protoMask = 0x80000000,
+    ProtoBuf = require('protobufjs'),
     Dota2 = exports;
 
-var Dota2Client = function Dota2Client(steamClient, debug, debugMore) {
+ProtoBuf.convertFieldsToCamelCase = true;
+var builder = ProtoBuf.newBuilder();
+ProtoBuf.loadProtoFile(path.join(__dirname, './proto/base_gcmessages.proto'), builder);
+ProtoBuf.loadProtoFile(path.join(__dirname, './proto/gcsdk_gcmessages.proto'), builder);
+ProtoBuf.loadProtoFile(path.join(__dirname, './proto/dota_gcmessages_client_fantasy.proto'), builder);
+ProtoBuf.loadProtoFile(path.join(__dirname, './proto/dota_gcmessages_client.proto'), builder);
+ProtoBuf.loadProtoFile(path.join(__dirname, './proto/dota_gcmessages_common.proto'), builder);
+ProtoBuf.loadProtoFile(path.join(__dirname, './proto/steammessages.proto'), builder);
+Dota2.schema = builder.build();
+
+var Dota2Client = function Dota2Client(steamUser, debug, debugMore) {
```

Comment:
I recommend not changing the API and continuing to pass a steamClient here.  You can construct a steamUser from a client.

```
var steamUser = new Steam.SteamUser(steamClient);
```